BookingSpecTests/SampleTest.feature

```diff
@@ -3,9 +3,122 @@
 	As a math idiot
 	I want to be told the sum of two numbers
 
-Scenario: Open Main Page and Click Sign In Button
+Scenario Template: Attempt to Sign In using Invalid combinations of email and password
 	Given I open browser
 	And I navigate to url "https://booking.com"
-	Then I wait for 5 seconds
 	When I click Sign In button
-	Then I wait for 5 seconds
\ No newline at end of file
+	And I set following parameters on Sign In Pop Up dialog
+		| Field    | Value      |
+		| Email    | <Email>    |
+		| Password | <Password> |
+	And I click Sign In button on PopUp
+	Then I wait while page popup is working
+	And I see message error with "<Message>"
+
+Scenarios: 
+		| Email                      | Password  | Message                                                               |
+		| testmirantistest@gmail.com | wrongpass | You entered an email address/password combination that doesn't match. |
+		| invalidEmail@g.com         | 1234qweR  | You entered an email address/password combination that doesn't match. |
+		| invalidEmail@g.com         | wrongpass | You entered an email address/password combination that doesn't match. |
+		|                            | 1234qweR  | Please enter a valid email address.                                   |
+		| testmirantistest@gmail.com |           | Please add a password                                                 |
+		|                            |           | Please enter a valid email address.                                   |
+
+	Scenario: Input valid credintials to Sign In
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	And I write email on Sign in PopUp "testmirantistest@gmail.com"
+	And I write password "1234qweR"
+	And I click Sign In button on PopUp
+	Then I see that i am Signed In
+
+Scenario: Input in email field more than 80 chars and assert that there only 80 chars
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	And I write email with length ninety chars "testmirantistesttestmirantistesttestmirantistesttestmirantistesttestmirantistest@gmail.ru"
+	Then I check that length of email is 80 chars
+
+Scenario: Attempt to recover password with invalid email
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	When I click Forgot Your Password button
+	And I write email "estmirantistest@gmail.com"
+	And I Click button Send
+	Then I wait while page popup is loading
+	Then I see an error on ForgotYourPasswordPopUp
```

Comment:
Need to avoid usage of such naming 'ForgotYourPasswordPopUp'.
Need to add message details into this step to make sure that user can see the correct error message.